common/objectcachev2.h

```diff
@@ -226,33 +220,21 @@ class ObjectCacheV2 {
 
     template <typename KeyType, typename Ctor>
     Borrow update(KeyType&& key, Ctor&& ctor) {
-        auto item = __find_or_create_item(std::forward<KeyType>(key));
-        auto r = __update(item, ctor());
-        return Borrow(this, item, std::move(r));
+        auto box = __find_or_create_box(std::forward<KeyType>(key));
+        DEFER(box.release());
+        auto r = ctor();
+        box.update(r, photon::now);
+        return Borrow(this, box, r);
     }
 
     ObjectCacheV2(uint64_t lifespan)
         : lifespan(lifespan),
           _timer(1UL * 1000 * 1000, {this, &ObjectCacheV2::__expire}, true,
-                 photon::DEFAULT_STACK_SIZE),
-          _exit(false) {
-        reclaimer = photon::thread_create11([this] {
-            while (!_exit) {
-                reclaim_queue.recv();
-            }
-        });
-        photon::thread_enable_join(reclaimer);
-    }
+                 photon::DEFAULT_STACK_SIZE) {}
 
     ~ObjectCacheV2() {
         _timer.stop();
-        if (reclaimer) {
-            _exit = true;
-            reclaim_queue.template send<PhotonPause>(nullptr);
-            photon::thread_join((photon::join_handle*)reclaimer);
-            reclaimer = nullptr;
-        }
         SCOPED_LOCK(maplock);
```

Comment:
use maplock only to protect a single assignment?

Reply:
code here want to access intrusive_list only for preventing hit assert of list.

Technically in dtor, other threads should not have any other accesses to the whole object cache, the lock here could be ignored.

I will remove the lock acquiring here